### DISCUSSION TITLE
Productionalizing R code

### DISCUSSION THREAD

**Discussion by rparmm** [rinpharma/rinpharma-summit-2024#16] [roundtable]
### Proposal

You created a small shiny application for your colleagues in a small team (2-5 people), that application is now getting extensive usage with more than 20, 30 or even 300 users. How do you go about scaling up your application? We want to hear from other users' experience in creating long lived R based workflows. This can be anything from small snippets of code that started off as scripts that can be sequestered into R packages that are idiomatic to commonly used R interfaces to cater to a wider audience. Shiny apps that are doing computationally heavy tasks with multiple users that require you to think about scalability, fault tolerance, and complexity.

### Expected impact

Sharing how others have made their R code into something that is much more user friendly or easier to contribute to. Code that readily interfaces with prior frameworks like tidyverse or scales easily. Other R users can navigate the trade-offs around consistency and scalability. Make informed decisions when writing R code taking into account future necessities such as validation requirements or maintainability.

### Prior discussions/work

Some prior work includes......
...

### Would you be willing to potentially facilitate this discussion?

None

**Comment by pzhang-cims** [comment]
@rparmm Thank you for raising this topic. We realize this dilemma as well. 
So Rshiny usually starts for a purpose of a tool, instantly interact for some small features. This is attractive as statistical programmer can just build UI and server to realize such purpose. - Later on some user might say “hey, can we another feature ,..", "Can this be changed to another range ...". With more request come in, documentation should be needed to play in an important role and a development cycle is needed, While a software development cycle can apply, e.g. from user journey to requirement to specification to UI/UX to OQ/PQ to validation to production, we might want to seek for a compatible workflow for Rshiny to be effective and efficient.

**Comment by nturaga** [comment]
Some thoughts in another discussion topic I started, that might be slightly related: https://github.com/rinpharma/rinpharma-summit-2024/discussions/22
